src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/ProviderBase/DbConnectionPool.cs

```diff
@@ -758,7 +758,7 @@ private DbConnectionInternal CreateObject(DbConnection owningObject, DbConnectio
                         }
                     }
                 }
-                SqlClientEventSource.Log.PoolerTraceEvent("<prov.DbConnectionPool.CreateObject|RES|CPOOL> {0}, Connection {1}, Added to pool.", ObjectID, newObj != null ? newObj?.ObjectID.ToString() ?? "null" : "null");
+                SqlClientEventSource.Log.PoolerTraceEvent("<prov.DbConnectionPool.CreateObject|RES|CPOOL> {0}, Connection {1}, Added to pool.", ObjectID, newObj != null ? newObj?.ObjectID ?? 0 : -1);
```

Comment:
the main Idea was avoiding unnecessary string creation. We can add to our Eventsource that value -1 is equal to null.

Reply:
Another solution to keep the `null` string would be checking the null objects outside the `PoolerTraceEvent` and use a specific message for it instead of passing a parameter.

Reply:
I thought about checking inside `PoolerTraceEvent` to see if the item is -1 add thenadd  "null" string, but because of the overrides we cannot say which one of the arguments is objectId.